integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java

```diff
@@ -0,0 +1,212 @@
+
+    assertThrows(
+        RetryWithBackoffException.class,
+        () -> eventBatchSender.sendBatch(eventBuffer.createBatch()));
+  }
+
+  private static Stream<Arguments> codesAndExpectedExceptions() {
+    return Stream.of(
+        //        Arguments.of(
+        //            413,
+        //            RetryWithSplitException.class,
+        //            "Batch was too large. Please try splitting and resending smaller sized
+        // batches."),
+        Arguments.of(
+            420,
+            RetryWithBackoffException.class,
+            "The New Relic API suggests backing off exponentially on this request."),
+        Arguments.of(
+            403,
+            DiscardBatchException.class,
+            "The New Relic API failed to process this request and it should not be retried."),
+        Arguments.of(
+            404,
+            DiscardBatchException.class,
+            "The New Relic API failed to process this request and it should not be retried."),
+        Arguments.of(429, RetryWithRequestedWaitException.class, "Please retry after 10 SECONDS"),
+        Arguments.of(
+            500,
+            RetryWithBackoffException.class,
+            "The New Relic API suggests backing off exponentially on this request."),
+        Arguments.of(
+            503,
+            RetryWithBackoffException.class,
+            "The New Relic API suggests backing off exponentially on this request."));
+  }
+
+  @ParameterizedTest
+  @MethodSource("codesAndExpectedExceptions")
+  @DisplayName("SDK responds appropriately to non-202s by suggesting a course of action.")
+  void testMetricSendResponseCodes(
+      int statusCode, Class<? extends ResponseException> exceptionClass, String exceptionMessage)
+      throws Exception {
+    mockServerClient
+        .when(
+            new HttpRequest()
+                .withMethod("POST")
+                .withPath("/v1/accounts/events")
+                .withHeader("Content-Type", "application/json; charset=utf-8")
+                .withHeader("Content-Length", ".*"))
+        .respond(new HttpResponse().withStatusCode(statusCode).withBody("" + statusCode));
+
+    long currentTimeMillis = 350;
+
+    EventBuffer eventBuffer = new EventBuffer(new Attributes());
+    eventBuffer.addEvent(
+        new Event("myEvent", new Attributes().put("key1", "val1"), currentTimeMillis));
+    eventBuffer.addEvent(
+        new Event("myEvent", new Attributes().put("key2", "val2"), currentTimeMillis + 1));
+    assertThrows(
+        exceptionClass,
+        () -> eventBatchSender.sendBatch(eventBuffer.createBatch()),
+        exceptionMessage);
+  }
+
+  //
+  // [{\"common\":{\"attributes\":{\"key1\":\"val1\"}},\"metrics\":[{\"name\":\"myEvent\",\"type\":\"count\",\"value\":1.0,\"timestamp\":1557180766612,\"interval.ms\":42,\"attributes\":{\"key2\":\"val2\"}}]}]"
+
+  private static class EventPayload {
+    private final List<Map<String, Object>> metrics;
```

Comment:
this is weird. why is there a `metrics` member in here.

Reply:
I suspect that the payload isn't being inspected at all by mockserver, due to the `ONLY_MATCHING_FIELDS` option. We should get this payload (and the comment) whipped into shape to make sure that we're actually testing the shape of the payload, and not just the headers, etc.

Reply:
This is exactly the same as the Span and Metrics case.

Reply:
It's not the same. In those cases, the payload structure matches the JSON (`common` + `metrics/spans`). This payload structure is different.

Reply:
I think when lombok got removed, the lack of getters on these little classes made it so we stopped verifying the payload structure. We need that back.

Reply:
I put in a PR to fix this for metrics/spans.  https://github.com/newrelic/newrelic-telemetry-sdk-java/pull/152